src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,62 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidRegistrationDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkForNulls(user);
+        checkLogin(user);
+        checkAge(user);
+        checkPassword(user);
+        checkIfUserExist(user);
```

Comment:
it is possible to first check whether the user already exists, before checking the validity of the fields (except for logins)